source/fundamentals/data-formats/document-data-format-pojo.txt

```diff
@@ -13,17 +13,226 @@ Document Data Format: POJOs
+   ``CodecRegistry`` allows you to specify one or more codec providers to
+   encode the POJO data. In this example, we call the following methods:
+
+   - :java-docs:`fromRegistries() <apidocs/bson/org/bson/codecs/configuration/CodecRegistries.html#fromRegistries(org.bson.codecs.configuration.CodecRegistry...)>`
+     to combine multiple ``CodecRegistry`` instances into a single one
+   - :java-docs:`getDefaultCodecRegistry() <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html?is-external=true#getDefaultCodecRegistry()>`
+     to retrieve a ``CodecRegistry`` instance from a list of codec providers
+   - :java-docs:`fromProviders() <apidocs/bson/org/bson/codecs/configuration/CodecRegistries.html#fromProviders(org.bson.codecs.configuration.CodecProvider...)>`
+     to create a ``CodecRegistry`` instance from the ``PojoCodecProvider``
+
+   See the code below to see how to instantiate the ``CodecRegistry``:
+
+   .. code-block:: java
+
+      // ensure you use the following static imports above your class definition
+      import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+      import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+      import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+      // ...
+
+      CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+#. Configure our ``MongoClient``, ``MongoDatabase``, or ``MongoCollection``
+   instance to use the Codecs in the ``CodecRegistry``. You only need to
+   configure one of these. In this example, we set the ``CodecRegistry`` on a
+   ``MongoDatabase`` called ``sample_pojos`` using the ``withCodecRegistry()``
+   method.
+
+   .. code-block:: java
+
+      try (MongoClient mongoClient = MongoClients.create(uri)) {
+          MongoDatabase database = mongoClient.getDatabase("sample_pojos").withCodecRegistry(pojoCodecRegistry);
+          // ...
+      }
+
+#. Configure your ``MongoCollection`` instance to cast to your POJO class when
+   retrieving documents:
+
+   .. code-block:: java
+
+      MongoCollection<Flower> collection = database.getCollection("flowers", Flower.class);
+
+Once you have configured the ``MongoCollection`` instance above, you can:
+
+- save an instance of the POJO to the collection
+- retrieve instances of the POJO from a query on the collection
+
+The code below shows you how you can set data in an instance of a ``Flower``,
+save it in the collection, and retrieve the data in a ``List`` of ``Flower``
+objects.
+
+.. code-block:: java
+
+   Flower flower = new Flower("rose", false, 25.4f, Arrays.asList(new String[] {"red", "green"}));
+
+   // insert the instance
+   collection.insertOne(flower);
+
+   // return all documents in the collection
+   List<Flower> flowers = new ArrayList<Flower>();
+
+   MongoCursor<Flower> cursor = collection.find().iterator();
```

Comment:
Use either toArray method on FindIterable, or if you want to show the loop, use try-with-resources statement

Reply:
I updated this to use the `into()` method on the `FindIterable`. Is that what you meant by `toArray`? I wasn't able to find a `toArray()` method on `MongoIterable`.